### PR TITLE
Page to post an array of entries for migration

### PR DESCRIPTION
**Pull request by alanmoo**

### CODE REVIEW

File: public/migrate.html [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+      var data = evt.target.response;
+      try {
+        data = JSON.parse(data);
+        currentUser = data.user;
+        currentNonce = data.nonce;
+        currentCSRF = data.csrf_token;
+
+        // when we have a nonce, set up actual form handling
+        var btn = document.querySelector("#submit");
+        btn.onclick = submitForm
```
Comment (Pomax) [inline public/migrate.html]: tiny user improvement: when you click "submit", grey out the submit button again, so you can't restart the submission process?

**Comment by Pomax** [comment]
We probably want to streamline this a little bit so we can throw away old style code:

- enable the button when text is put into the textarea (*and* `JSON.parse` doesn't error on that input =)
- when clicking submit, convert text into an `entries` array
- start to iterate through the array:
  - doNextEntry() => entry = entries.pop(), get nonce
  - getNonce(entry) => handleNonceRequest
  - handleNonceRequest(entry, data) => prepare data payload based on entry + data.nonce etc
  - postData(data) => xhr with 'done' callback to doNextEntry

File: public/migrate.html [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+      entries = JSON.parse(document.getElementById("entryArray").value);
+      getNonce(postAnEntry);
+    };
+
+    // don't let the browser submit this form. We'll handle it.
+    document.forms.entries.onsubmit = () => false;
+    document.querySelector("#submit").onclick = submitForm;
+    var textBox = document.getElementById("entryArray");
+    var submitBtn = document.querySelector("#submit");
+    textBox.onkeyup = (e) => {
```
Comment (Pomax) [inline public/migrate.html]: will this trigger on a mouse right-click + paste?

Comment (alanmoo) [inline public/migrate.html]: No. What handler would you suggest? `onchange` seemed to depend on focus, so that wasn't great either.

**Comment by Pomax** [comment]
R+!

**Pull request merged**